src/progress.rs

```diff
@@ -824,6 +824,33 @@ impl ProgressBar {
         self.state.lock().unwrap().abandon_with_message(msg);
     }
 
+    /// Finishes the progress bar using the [`ProgressFinish`] behavior stored
+    /// in the [`ProgressStyle`].
+    pub fn finish_using_style(&self) {
+        let on_finish = self.state.lock().unwrap().style.get_on_finish().clone();
```

Comment:
Do we need the `.clone()` here?

Style nits: let's just inline the value of `on_finish` in the `match` statement below. I also think the match arms here could just be `self.finish(),` rather than `{\nself.finish();\n}`, can you see if that works? Additionally let's just skip the `/* do nothing */` comment, I don't think it adds value here.